Simulator/simulator/garden.py

```diff
@@ -279,14 +279,17 @@ def irrigate(self, location, amount):
 
         Args:
             location (Array of [int,int]): Location [row, col] where to perform actions.
-            amount (int) amount of water for location.
+            amount (float) amount of water for location.
 
         """
         lower_x = max(0, location[0] - self.irr_threshold)
         upper_x = min(self.grid.shape[0], location[0] + self.irr_threshold + 1)
         lower_y = max(0, location[1] - self.irr_threshold)
         upper_y = min(self.grid.shape[1], location[1] + self.irr_threshold + 1)
-        self.grid[lower_x:upper_x, lower_y:upper_y]['water'] += amount
+        window_grid_size = (self.irr_threshold+self.irr_threshold+1)*(self.irr_threshold+self.irr_threshold+1)/10000  # in square meters
+        # TODO: add distribution kernel for capillary action and spread of water jet
+        # 0.001m^3/(0.11m * 0.11m * 0.35m) ~ 0,236 %
+        self.grid[lower_x:upper_x, lower_y:upper_y]['water'] += amount/(window_grid_size*0.35)  # 0.0121m^2 * 0.35m depth
```

Comment:
If irr_threshold is 5, that would make window_grid_size ~= 0.04.  Then wouldn't the final amount on line 292 be very large?  If "amount" = 1, then the water added, before the minimum on the next line) would be 250?

Also seems like we can combine the np.minimum code and line 289 into one -> self.grid[][] = min(amount/window..., MAX_WATER_LEVEL)

Reply:
Want to achieve a certain VWC (vol water content in soil)
If irr_threshold is 5: 
window_grid_size = (11cm*11cm)/10000 = 0,0121m^2
assuming uniform distribution along depth of garden (35cm) water amount of water window_grid_size gets added to water grid.

Reply:
I'm still confused. Given amount = 1,
if irr_threshold is 5:
window_grid_size = (11*11)/10000 = 0.0121 m^2
self.grid[]['water'] += 1 / (0.0121*0.35)

If 1 / (0.0121*0.35) = 236 then the amount of water we are putting into each cell is enormous.  Maybe I'm not understanding the volume changes and we can hop on a call Friday.